### PR TITLE
Runtime: Core BPF Migration: Struct for loading and checking builtin program accounts

### PR DESCRIPTION
**Pull request by buffalojoec**
This is chunk 3/7 of the broken-up PR #79.

#### Problem
When the runtime desires to migrate a builtin program to Core BPF, there exists
no reliable method for loading a builtin's program account and performing
various checks to ensure the program account is configured as expected.

Being able to run these checks before fetching the builtin's program account
is extremely important for ensuring a smooth migration to Core BPF.

#### Summary of Changes
Within the `builtins` submodule, add a module specifically for `core_bpf_migration`,
and inside of it create a new structure for loading a builtin's program account and
running checks, based on the type of builtin.

### CODE REVIEW

**Comment by buffalojoec** [comment]
@CriesofCarrots How best to stack on #304? Wait for merge, or create branch in the monorepo? Seems like no one else has any custom branches in there at the moment.

**Comment by buffalojoec** [comment]
Note this can go in separate from #318.

File: runtime/src/bank/builtins/core_bpf_migration/mod.rs
Comment (CriesofCarrots) [inline runtime/src/bank/builtins/core_bpf_migration/mod.rs]:
```suggestion
pub(crate) enum CoreBpfMigrationTargetType {
```

File: runtime/src/bank/builtins/core_bpf_migration/target_builtin.rs
Comment (CriesofCarrots) [inline runtime/src/bank/builtins/core_bpf_migration/target_builtin.rs]: Recommend removing this line (dereference) to avoid a potentially unnecessary copy, since the only non-error users of this variable before constructing the new struct take a reference.
If you want this to be called `program_address`, update it in the method signature.

Comment (buffalojoec) [reply runtime/src/bank/builtins/core_bpf_migration/target_builtin.rs]: Sounds good I've updated this function.

**Review by CriesofCarrots — APPROVED** [review]
lgtm assuming CI is happy

**Pull request merged**
